run.py

```diff
@@ -410,31 +464,71 @@ def run_diffusion_processsing(**args):
                 if stage in args.stages:
                     stage_func()
 
-        dwis = layout.get(subject=subject_label, type='dwi',
-                                                 extensions=["nii.gz", "nii"])
+        # dwis = layout.get(subject=subject_label, type='dwi',
+        #                                          extensions=["nii.gz", "nii"])
 
         # print(dwis)
-        # acqs = set(layout.get(target='acquisition', return_type='id',
-        #                       subject=subject_label, type='dwi',
-        #                       extensions=["nii.gz", "nii"]))
-        # print(acqs)
-        # posData = []
-        # negData = []
-        # for acq in acqs:
-        #     pos = "EMPTY"
-        #     neg = "EMPTY"
-        #     dwis = layout.get(subject=subject_label,
-        #                       type='dwi', acquisition=acq,
-        #                       extensions=["nii.gz", "nii"])
-        #     assert len(dwis) <= 2
-        #     for dwi in dwis:
-        #         dwi = dwi.filename
-        #         if "-" in layout.get_metadata(dwi)["PhaseEncodingDirection"]:
-        #             neg = dwi
-        #         else:
-        #             pos = dwi
-        #     posData.append(pos)
-        #     negData.append(neg)
-        #
-        # print(negData)
-        # print(posData)
+
+        posData = []
+        negData = []
+        PEdir = "None"
+        dwiname = "Diffusion"
+        dirnums = []
+
+        numruns = set(layout.get(target='run', return_type='id',
+                                 subject=subject_label, type='dwi',
+                                 extensions=["nii.gz", "nii"]))
+
+        for session in numruns:
+            acqs = set(layout.get(target='acquisition', return_type='id',
+                                  subject=subject_label, type='dwi',
+                                  extensions=["nii.gz", "nii"]))
+            for acq in acqs:
+                y = [int(s) for s in acq[0:len(acq)] if s.isdigit()]
+                y = [str(s) for s in y]
+                y = ''.join(y)
+                dirnums.append(y)
+            for dirnum in set(dirnums):
+                dwiname = "Diffusion" + "_dir-" + dirnum + "_" + session + "_corr"
+
+                diracqs = [x for x in acqs if dirnum in x]
+                if "AP" or "PA" in diracqs:
```

Comment:
You need to use the metadata in the sidecar JSON file instead of file names - we cannot rely on those. You can easily access it via get_metadata method from pybids.